mappings/doab.py

```diff
@@ -2,129 +2,179 @@
+            return None
+
+        title = doab_record.xpath('./dc:title/text()', namespaces=namespaces) + doab_record.xpath('./datacite:creator/text()', namespaces=namespaces)
+        relations = doab_record.xpath('./dc:relation/text()', namespaces=namespaces)
+        publishers = doab_record.xpath('./dc:publisher/text()', namespaces=namespaces)
+        languages = doab_record.xpath('./dc:language/text()', namespaces=namespaces)
+        extents = doab_record.xpath('./oapen:pages/text()', namespaces=namespaces)
+        
+        return Record(
+            uuid=uuid4(),
+            frbr_status=FRBRStatus.TODO.value,
+            cluster_status=False,
+            source=Source.DOAB.value,
+            source_id=source_id,
+            identifiers= identifiers,
+            authors=self._get_authors(doab_record, namespaces=namespaces),
+            contributors=self._get_contributors(doab_record, namespaces=namespaces),
+            title=title[0] if len(title) > 0 else None,
+            is_part_of=[f'{part}||series' for part in relations],
+            publisher=[f'{publisher}||' for publisher in publishers],
+            spatial=doab_record.xpath('./oapen:placepublication/text()', namespaces=namespaces),
+            dates=self._get_dates(doab_record, namespaces=namespaces),
+            languages=[f'||{language}' for language in languages],
+            extent=[f'{extent} pages' for extent in extents],
+            abstract=doab_record.xpath('./dc:description/text()', namespaces=namespaces),
+            subjects=self._get_subjects(doab_record, namespaces=namespaces),
+            has_part=self._get_has_part(doab_record, namespaces),
+            rights=self._get_rights(doab_record, namespaces=namespaces),
+            date_created=datetime.now(timezone.utc).replace(tzinfo=None),
+            date_modified=datetime.now(timezone.utc).replace(tzinfo=None)
+        )
+
+    def _get_text_type_data(self,record, namespaces, field_xpath, format_string):
+        field_data = record.xpath(field_xpath, namespaces=namespaces)
+
+        if not field_data:
+            return None
+
+        data_tuples = [
+            (item.text, item.get('type') or '')
+            for item in field_data
+        ]
+        
+        return [format_string.format(item[0], item[1]) for item in data_tuples]
 
-        # Clean up subjects
-        self.record.subjects = list(filter(lambda x: x[:3] != 'bic', self.record.subjects))
+    def _get_authors(self, record, namespaces):
+        datacite_authors = record.xpath('./datacite:creator/text()', namespaces=namespaces)
+        dc_authors = record.xpath('./dc:creator/text()', namespaces=namespaces)
 
-        # Clean up rights statements
-        self.record.rights = self.parseRights()
+        authors = datacite_authors or dc_authors
 
-        # Clean up links
-        self.record.has_part = self.parseLinks()
+        return [f'{author}|||true' for author in authors]
```

Comment:
We may want to add `if author is not None or author == ''` to ensure we aren't adding empty strings